### PR TITLE
Extract conditions

### PR DESCRIPTION
**Pull request by collerek**
Refactors group of conditions into the named token functions to increase the readability of the parser (it was hard to grasp what is checked in a given place with so many chained conditions) as well as fixes code complication issues.

Close: #175, close: #176, close: #177

### CODE REVIEW

**Comment by collerek** [comment]
That might seem like a lot of changes but it's just moving the logic from the parser to token and to helper private methods, there is almost nothing new (except one fix after columns method simplification)

**Comment by collerek** [comment]
@macbre any ETA on review of this PR? I can probably tackle some of the bugs reported in issues but one PR at a time :)

**Comment by macbre** [comment]
@collerek, good point, I'm on it 🙂

File: sql_metadata/conditions.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
+"""
+Gathers conditions used to parse tokens into specified qualifiers
+"""
```
Comment (macbre) [inline sql_metadata/conditions.py]: 👌🏻

**Review by macbre — APPROVED** [review]

**Pull request merged**